### PR TITLE
Annotate a CSV with plate/well metadata

### LINKED ISSUE #13: Implement annotate
Convert the [annotate script](https://github.com/broadinstitute/cytominer_scripts/blob/master/annotate.R) to an R function.

### PR DESCRIPTION
**Pull request by mcquin**
Resolves #13

### CODE REVIEW

**Comment by shntnu** [comment]
I used `.data` to overcome the "no visible bindings issue"
https://stackoverflow.com/questions/9439256/how-can-i-handle-r-cmd-check-no-visible-binding-for-global-variable-notes-when
I previously had used `rlang::sym` to overcome this
https://github.com/cytomining/cytominer/blob/649ba287f7f8ff7faa0992fba276ebee1d22354f/R/drop_na_columns.R#L35-L39
but changed my mind after reading 

https://community.rstudio.com/t/should-tidyeval-be-abandoned/
especially
https://community.rstudio.com/t/should-tidyeval-be-abandoned/2238/5

I find tidyeval very helpful but it does cause some headaches. I anticipate this getting easier over the next year once it matures.

**Review by shntnu — APPROVED** [review]

**Comment by mcquin** [comment]
👏 @shntnu!

File: R/annotate.R [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+#' - \code{<workspace_dir>/metadata/barcode_platemap.csv} which maps the plate
+#' barcodes to the corresponding platemaps,
+#' - For each platemap, a file
+#' \code{<workspace_dir>/metadata/platemap/<plate_map_name>.txt} which
+#' contains the metadata for each well position
+#'
+#' @param batch_id            Batch ID.
+#' @param plate_id            Plate ID.
+#' @param cell_id             Optional cell ID. default: \code{NULL}.
+#' @param external_metadata   Optional external metadata to join with, as a JSON file. default: \code{NULL}.
```
Comment (shntnu) [inline R/annotate.R]: as a CSV file, not JSON

**Review by shntnu — COMMENTED** [review]
@mcquin thanks for adding that test!

File: tests/testthat/test-annotate.R [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+  file.remove(annotated_csv)
+})
+
+test_that("`annotate` with external_metadata appends metadata", {
+  annotated_csv <- tempfile("SQ00015116_augmented.csv")
+
+  external_metadata_csv <- tempfile("external_m")
+
+  external_metadata <- tibble(
```
Comment (shntnu) [inline tests/testthat/test-annotate.R]: `dplyr::tibble`

File: tests/testthat/test-annotate.R [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+  file.remove(external_metadata_csv)
+})
+
+test_that("`annotate` with external_metadata adds metadata prefix", {
+  annotated_csv <- tempfile("SQ00015116_augmented.csv")
+
+  external_metadata_csv <- tempfile("external_m")
+
+  external_metadata <- tibble(
```
Comment (shntnu) [inline tests/testthat/test-annotate.R]: `dplyr::tibble`

**Review by shntnu — CHANGES_REQUESTED** [review]
Looks good except for two changes suggested. Awesome that we have a test for external metadata!

**Review by shntnu — COMMENTED** [review]
Minor fix for docs, looks good @mcquin !

**Review by shntnu — APPROVED** [review]

**Pull request merged**